src/components/Modal/Modal.tsx

```diff
@@ -88,11 +115,11 @@ class Modal extends PureComponent {
     return createPortal(
       <div
         className={cn("modal-overlay fixed inset-0 bg-black-o-75 z-50", {
-          "opacity-0": this.isAboutToClose
+          "opacity-0": isAboutToClose
         })}
-        onMouseUp={this.gracefulClose}
+        onMouseUp={(e) => this.gracefulClose(e)}
         onMouseDown={() => {
-          this.isMouseClicked = true;
+          this.setState({ isMouseClicked: true });
```

Comment:
**thought** So the thing is, here we set the state but since it's an async operation, it may occur that the state is not updated between the time `onMouseDown` and `onMouseUp` are executed. 

So probably this happens:

`onMouseDown` => `onMouseUp` (this reads `isMouseClicked` as false) => state update 

instead we want that:

`onMouseDown` => update the flag => `onMouseUp` (so that isMouseClicked is `true`)

Reply:
I have done some tests and the solutions could be this:
```
async() =>
          this.setState({}, () => this.state.isMouseClicked = true)
        }
```
But yeah, I think it's too articulated so I revert the operation to the original